src/ecdsa/ecdsa.py

```diff
@@ -171,7 +171,13 @@ def sign(self, hash, random_k):
     G = self.public_key.generator
     n = G.order()
     k = random_k % n
-    p1 = k * G
+    # Fix the bit-length of the random nonce,
+    # so that it doesn't leak via timing.
+    # This does not change that ks = k mod n
+    ks = k + n
+    if ks.bit_length() == n.bit_length():
```

Comment:
ints don't have `bit_length()` on python2.6, you can use `bit_length` from `rfc6979` module instead